### PR TITLE
layers: Fix decompression layer crash

### PR DESCRIPTION
**Pull request by vkushwaha-nv**
This happened when subgroupsize is 16

Also fix the test to use BDA usage

Worth checking if this passes on Pixel 7 as well. 

This fix was suggested by ARM which has similar shader params.

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
This passes on Pixel 7:

```
[----------] 2 tests from DecompressionTest
[ RUN      ] DecompressionTest.DecompressMemory
[       OK ] DecompressionTest.DecompressMemory (710 ms)
[ RUN      ] DecompressionTest.DecompressMemoryIndirect
[       OK ] DecompressionTest.DecompressMemoryIndirect (617 ms)
[----------] 2 tests from DecompressionTest (1328 ms total)
```
Galaxy S23 is still crashing:
----------] 2 tests from DecompressionTest
[ RUN      ] DecompressionTest.DecompressMemory
```
/home/lunarg/.jenkins/wvel/Debug64/Vulkan-ExtensionLayer/tests/vktestbinding.cpp:281: Failure
Expected equality of these values:
  (result = vk::CreateDevice(phy_.handle(), &info, __null, &dev))
    Which is: -13
  VK_SUCCESS
    Which is: 0

/home/lunarg/.jenkins/wvel/Debug64/Vulkan-ExtensionLayer/tests/vkrenderframework.cpp:941: Failure
Expected: (true) != (graphics_queues().empty()), actual: true vs true
```

File: layers/decompression/decompression.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * Author: Vikram Kushwaha <vkushwaha@nvidia.com>
  */
 
-static bool logging_enabled = false;
+static bool logging_enabled = true;
```
Comment (jeremyg-lunarg) [inline layers/decompression/decompression.cpp:20]: I didn't see any logging in the test run output.

**Review by jeremyg-lunarg — COMMENTED** [review]
I think we should merge this but you need to turn logging_enabled back off.  Then we can enable these tests on Pixel 7 by default. Galaxy S23 still has a different problem.

**Comment by juan-lunarg** [comment]
FWIW this fixes testing on MoltenVK as well.

File: layers/decompression/decompression.cpp (hunk 1 above)
Comment (juan-lunarg) [reply layers/decompression/decompression.cpp:20]: I didn't see any testing either.

File: tests/decompression_tests.cpp
Comment (juan-lunarg) [inline tests/decompression_tests.cpp:1]: Fixes MoltenVK tested on an M2

```
if (InstanceExtensionSupported(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME)) {
    GTEST_SKIP() << "VK_KHR_portability_subset enabled, skipping.\n";
}
```

Remove code that skips if it detects the `VK_KHR_portability_subset`

**Comment by juan-lunarg** [comment]
Also for what it's worth @vkushwaha-nv the binaries are built as Debug on Android CI now. Which improves the logging when trying to figure out Android issues.

**Comment by vkushwaha-nv** [comment]
Thanks all. I have disabled logging in the second commit.
The failure on Galaxy S23 at device creation time is ERROR_UKNOWN, not very helpful unfortunately. We probably need someone from Samsung to help on why device creation fails, this is about enabling features/extensions I think and not actual decompression.

Hi Juan, can you enable this MoltenVK too?

**Comment by juan-lunarg** [comment]
> Hi Juan, can you enable this MoltenVK too?

👍🏾

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**